contracts/DAOTreasury.cdc

```diff
@@ -196,6 +255,61 @@ pub contract DAOTreasuryV2 {
       self.depositCollection(collection: <- collection)
     }
 
+    pub fun signerRemoveCollection(identifier: String, signaturePayload: MyMultiSigV2.MessageSignaturePayload) {
+      pre {
+        self.collections[identifier] != nil: "Collection doesn't exist in this treasury."
+        self.collections[identifier]?.getIDs()?.length == 0 : "Collection must be empty before it can be removed."
+      }
+      // ------- Validate Address is a Signer on the Treasury -----
+      let signers = self.multiSignManager.getSigners()
+      assert(signers[signaturePayload.signingAddr] == true, message: "Address is not a signer on this Treasury")
+
+      // ------- Validate Message --------
+      // message format: {collection identifier hex}{blockId}
+      var counter = 0
+      let signingBlock = getBlock(at: signaturePayload.signatureBlock)!
+      let blockId = signingBlock.id
+      let blockIds: [UInt8] = []
+
+      while (counter < blockId.length) {
+          blockIds.append(blockId[counter])
+          counter = counter + 1
+      }
+
+      let blockIdHex = String.encodeHex(blockIds)
+      let collectionIdHex = String.encodeHex(identifier.utf8)
+
+      let message = signaturePayload.message
+
+      // Collection Identifier
+      assert(
+        collectionIdHex == message.slice(from: 0, upTo: collectionIdHex.length),
+        message: "Invalid Message: incorrect collection identifier"
+      )
+      // Block ID
+      assert(
+        blockIdHex == message.slice(from: collectionIdHex.length, upTo: message.length),
+        message: "Invalid Message: invalid blockId"
+      )
+
+      // ------ Validate Signature -------
+      var signatureValidationResponse = MyMultiSigV2.validateSignature(payload: signaturePayload)
+
+      assert(
+        signatureValidationResponse.isValid == true,
+        message: "Invalid Signature"
+      )
+      assert(
+        signatureValidationResponse.totalWeight >= 999.0,
+        message: "Insufficient Key Weights: sum of total signing key weights must be >= 999.0"
+      )
```

Comment:
This part is similar (or maybe same) for all signerFunctions. Maybe this part can be extracted to a function (it can be private), plus with all those FCLCrypto contract changes we're just gonna change this part from 296-305 lines

Reply:
I refactored this a bit, and split out the blockId validation into its own function.  Not going to consolidate all of it, though, as I said in Slack, to give us flexibility to change message format in the future.